src/libs/antares/timeelapsed/timer.h

```diff
@@ -28,38 +28,43 @@
 #define __ANTARES_LIBS_TIME_ELAPSED__TIME_ELAPSED_H__
 
 #include <yuni/yuni.h>
-#include <yuni/core/string.h>
+#include <yuni/string.h>
 
-class TimeElapsed final
+namespace TimeElapsed
+{
+class Aggregator;
+
+class Timer final
 {
 public:
     //! Default constructor
-    TimeElapsed();
+    Timer() = delete;
 
     //! Copy constructor
-    TimeElapsed(const TimeElapsed& rhs) : text(rhs.text), pStartTime(rhs.pStartTime)
-    {
-    }
+    Timer(const Timer& rhs) = delete;
+    Timer& operator=(const Timer& rhs) = delete;
 
     //! Constructor with a default text
-    explicit TimeElapsed(const AnyString& text) : text(text)
-    {
-        reset();
-    }
-
-    //! Destructor
-    ~TimeElapsed();
+    explicit Timer(const AnyString& userText,
+                   const AnyString& logText = "",
+                   bool verbose = true,
+                   Aggregator* aggregator = nullptr);
 
     void reset();
-
-public:
-    //! The text to display at the end
-    Yuni::ShortString256 text;
+    void tick();
 
 private:
+    //! The text to display at the end
+    Yuni::ShortString256 userText;
+    // Aggregation
+    Yuni::ShortString256 logText;
     //! timestamp when the timer started
     yint64 pStartTime;
-
-}; // class TimeElapsed
+    //! Display text on destruction
+    bool verbose;
+    //! Aggregation into a single file for automation
+    Aggregator* pAggregator;
```

Comment:
Here we have a simple pointer as a data member. It reminds me of another situation where we made it a smart or rather weak pointer...

Reply:
True, but the aggregator is declared at the very top (application), as a stack variable. IMO there is no risk of memory leak or early de-allocation.